src/frontend/index.ts

```diff
@@ -184,57 +222,70 @@ export async function createFlowrWindow(flowrStore: Store<IFlowrStore>): Promise
 
       evt.sender.send('receiveConfig', config)
     },
-    getMacAddress: async (evt: any) => {
+    getErrorLoadingFlowr: (evt: IpcMainEvent) => {
+        const errorData: any = {
+          reloadTimeout: reloadTimeout._idleStart as number,
+          reloadInterval: reloadTimeout._idleTimeout as number,
+          url: flowrStore.get('extUrl') || defaultUrl,
+          lastError: lastError
+          }
+        evt.sender.send('receiveErrorLoadingFlowr', errorData)
+    },
+    getMacAddress: async (evt: IpcMainEvent) => {
       const activeMacAddress = await getActiveMacAddress()
       evt.sender.send('receiveMacAddress', activeMacAddress)
     },
-    getActiveMacAddress: async (evt: any) => {
+    getActiveMacAddress: async (evt: IpcMainEvent) => {
       const activeMacAddress = await getActiveMacAddress()
       evt.sender.send('receiveActiveMacAddress', activeMacAddress)
     },
-    getAllMacAddresses: async (evt: any) => {
+    getAllMacAddresses: async (evt: IpcMainEvent) => {
       try {
         const allMacAddresses = await getAllMacAddresses()
         evt.sender.send('receiveAllMacAddresses', allMacAddresses)
       } catch (e) {
         evt.sender.send('receiveAllMacAddresses', [])
       }
     },
-    getIpAddress: async (evt: any) => {
+    getIpAddress: async (evt: IpcMainEvent) => {
       try {
         const ipAddress = await getIpAddress()
         evt.sender.send('receiveIpAddress', ipAddress)
       } catch (e) {
         evt.sender.send('receiveIpAddress', '127.0.0.1')
       }
     },
-    updateAppConfig: (evt: any, data: any) => {
+    updateAppConfig: (evt: IpcMainEvent, data: IFlowrConfig) => {
       const currentConfig = flowrStore.get('flowrConfig')
       const newConfig = deepExtend(currentConfig, data)
       flowrStore.set('flowrConfig', newConfig)
       app.relaunch()
       app.quit()
     },
-    setDebugMode: (evt: any, debugMode: boolean) => {
+    reload:() => {
+      reload()
+      reloadTimeout = setInterval(reload, RELOAD_INTERVAL)
```

Comment:
This is not your doing, but what bothers me a lot is that we apply an interval in a variable called timeout.